src/htmlminifier.js

```diff
@@ -941,336 +964,425 @@ function minify(value, options, partialMarkup) {
+                    tagIndex--;
+                  }
+                  trimTrailingWhitespace(tagIndex - 1, 'br');
+                }
               }
+              else if (inlineTextTags(prevTag.charAt(0) === '/' ? prevTag.slice(1) : prevTag)) {
+                text = collapseWhitespace(text, options, /(?:^|\s)$/.test(currentChars));
+              }
+            }
+            if (prevTag || nextTag) {
+              text = collapseWhitespaceSmart(text, prevTag, nextTag, options);
+            }
+            else {
+              text = collapseWhitespace(text, options, true, true);
+            }
+            if (!text && /\s$/.test(currentChars) && prevTag && prevTag.charAt(0) === '/') {
+              trimTrailingWhitespace(buffer.length - 1, nextTag);
             }
           }
-          if (prevTag) {
-            if (prevTag === '/nobr' || prevTag === 'wbr') {
-              if (/^\s/.test(text)) {
-                var tagIndex = buffer.length - 1;
-                while (tagIndex > 0 && buffer[tagIndex].lastIndexOf('<' + prevTag) !== 0) {
-                  tagIndex--;
+          if (!stackNoCollapseWhitespace.length && nextTag !== 'html' && !(prevTag && nextTag)) {
+            text = collapseWhitespace(text, options, false, false, true);
+          }
+        }
+        if (options.processScripts && specialContentTags(currentTag)) {
+          text = processScript(text, options, currentAttrs);
+        }
+
+        var runningAsyncTask = false;
+
+        /**
+         * To be called when running an async task.
+         *
+         * @param {AsyncTextPlaceholder} placeholder
+         */
+        function onAsyncTask(placeholder) {
+          buffer.push(placeholder);
+          runningAsyncTask = true;
+          asyncTasksWaitingFor++;
+        }
+
+        /**
+         * Returns the callback function for an async task.
+         *
+         * @param {AsyncTextPlaceholder} placeholder
+         * @returns {Function}
+         */
+        function getAsyncTaskCallback(placeholder) {
+          return function(result) {
+            if (!runningAsyncTask) {
+              throw new Error('Async completion has already occurred.');
+            }
+            runningAsyncTask = false;
+            placeholder.setValue(result);
+            if (--asyncTasksWaitingFor === 0) {
+              if (typeof cb === 'function') {
+                if (error) {
+                  cb(error, null);
+                }
+                else {
+                  cb(null, finalize());
                 }
-                trimTrailingWhitespace(tagIndex - 1, 'br');
               }
             }
-            else if (inlineTextTags(prevTag.charAt(0) === '/' ? prevTag.slice(1) : prevTag)) {
-              text = collapseWhitespace(text, options, /(?:^|\s)$/.test(currentChars));
-            }
+          };
+        }
+
+        if (isExecutableScript(currentTag, currentAttrs)) {
+          var minifyJSPlaceholder = new AsyncTextPlaceholder();
```

Comment:
Clever use of `buffer` - unlike custom fragments, scripts and style sheets are guaranteed to take up the whole `text` :+1: 

May I suggest storing the index, i.e. `buffer.length` then push `null` or whatever to it, instead of introducing a new class?

Reply:
Done